#!/usr/bin/env python
PACKAGE = 'recast_ros'
import roslib; roslib.load_manifest(PACKAGE)


from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("cell_size",    double_t,    0, "Cell Size", 0.1,  0.015, 1)
gen.add("cell_height",    double_t,    0, "Cell Height", 0.1,  0.01, 1)
gen.add("agent_height",    double_t,    0, "Agent Height", 0.7,  0.01, 5)
gen.add("agent_radius",    double_t,    0, "Agent Radius", 0.2,  0, 5)
gen.add("agent_max_climb",    double_t,    0, "Max Climbing", 0.41,  0, 5)
gen.add("agent_max_slope",    double_t,    0, "Max Slope", 60.0,  0, 90)
gen.add("TERRAIN_TYPE0_COST",    double_t,    0, "Area Type0 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE1_COST",    double_t,    0, "Area Type1 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE2_COST",    double_t,    0, "Area Type2 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE3_COST",    double_t,    0, "Area Type3 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE4_COST",    double_t,    0, "Area Type4 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE5_COST",    double_t,    0, "Area Type5 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE6_COST",    double_t,    0, "Area Type6 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE7_COST",    double_t,    0, "Area Type7 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE8_COST",    double_t,    0, "Area Type8 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE9_COST",    double_t,    0, "Area Type9 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE10_COST",    double_t,    0, "Area Type10 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE11_COST",    double_t,    0, "Area Type11 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE12_COST",    double_t,    0, "Area Type12 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE13_COST",    double_t,    0, "Area Type13 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE14_COST",    double_t,    0, "Area Type14 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE15_COST",    double_t,    0, "Area Type15 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE16_COST",    double_t,    0, "Area Type16 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE17_COST",    double_t,    0, "Area Type17 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE18_COST",    double_t,    0, "Area Type18 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE19_COST",    double_t,    0, "Area Type19 Cost", 1.0,  0, 100.0)
gen.add("TERRAIN_TYPE20_COST",    double_t,    0, "Area Type20 Cost", 1.0,  0, 100.0)

exit(gen.generate(PACKAGE, "recast_ros", "recast_node"))
